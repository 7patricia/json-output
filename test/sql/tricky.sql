\set ECHO none
SET client_min_messages TO error;
\t on
\x off
SELECT 'created logical replication slot'
  FROM pg_create_logical_replication_slot('json', 'json-output');

BEGIN;

CREATE SCHEMA IF NOT EXISTS json;
SET LOCAL search_path TO json, public;
CREATE EXTENSION IF NOT EXISTS hstore;

CREATE TABLE IF NOT EXISTS tab1 (
  i   integer PRIMARY KEY,
  h   hstore NOT NULL DEFAULT ''
);

CREATE TABLE IF NOT EXISTS tab2 (
  i   integer PRIMARY KEY,
  j   integer NOT NULL DEFAULT 0
);

INSERT INTO tab1 (i) VALUES (1);
INSERT INTO tab2 (i) VALUES (1);

INSERT INTO tab1 (i) VALUES (2);
INSERT INTO tab2 (i) VALUES (2);

INSERT INTO tab1 (i) VALUES (3);
INSERT INTO tab2 (i) VALUES (3);

INSERT INTO tab1 (i) VALUES (4);
INSERT INTO tab2 (i) VALUES (4);

CREATE VIEW changedata AS
SELECT data FROM pg_logical_slot_get_changes('json', NULL, NULL);

END;

BEGIN;
SET LOCAL search_path TO json, public;
UPDATE tab1 SET h = hstore('i', i::text)||hstore('2i', (2*i)::text);
DELETE FROM tab1 WHERE i % 2 = 1;
DELETE FROM tab2 WHERE i % 2 = 1;
END;

SELECT * FROM json.changedata;

SELECT 'deleted logical replication slot'
  FROM pg_drop_replication_slot('json');

DROP SCHEMA json CASCADE;
