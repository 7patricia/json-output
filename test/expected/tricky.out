\set ECHO none
 created logical replication slot

 {"schema":"json","table":"tab1","action":"insert","new":{"i":1,"h":{}}}
 {"schema":"json","table":"tab2","action":"insert","new":{"i":1,"j":0}}
 {"schema":"json","table":"tab1","action":"insert","new":{"i":2,"h":{}}}
 {"schema":"json","table":"tab2","action":"insert","new":{"i":2,"j":0}}
 {"schema":"json","table":"tab1","action":"insert","new":{"i":3,"h":{}}}
 {"schema":"json","table":"tab2","action":"insert","new":{"i":3,"j":0}}
 {"schema":"json","table":"tab1","action":"insert","new":{"i":4,"h":{}}}
 {"schema":"json","table":"tab2","action":"insert","new":{"i":4,"j":0}}
 {"schema":"json","table":"tab1","action":"update","new":{"i":1,"h":{"i": "1", "2i": "2"}}}
 {"schema":"json","table":"tab1","action":"update","new":{"i":2,"h":{"i": "2", "2i": "4"}}}
 {"schema":"json","table":"tab1","action":"update","new":{"i":3,"h":{"i": "3", "2i": "6"}}}
 {"schema":"json","table":"tab1","action":"update","new":{"i":4,"h":{"i": "4", "2i": "8"}}}
 {"schema":"json","table":"tab1","action":"delete","old":{"i":1,"h":null}}
 {"schema":"json","table":"tab1","action":"delete","old":{"i":3,"h":null}}
 {"schema":"json","table":"tab2","action":"delete","old":{"i":1,"j":null}}
 {"schema":"json","table":"tab2","action":"delete","old":{"i":3,"j":null}}

 deleted logical replication slot

